cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedSparkProgramRunner.java

```diff
@@ -76,14 +78,18 @@ protected ProgramController launch(Program program, ProgramOptions options,
     SparkSpecification spec = appSpec.getSpark().get(program.getName());
     Preconditions.checkNotNull(spec, "Missing SparkSpecification for %s", program.getId());
 
-    // Localize the spark-assembly jar
-    File sparkAssemblyJar = SparkUtils.locateSparkAssemblyJar();
-    localizeResources.put(sparkAssemblyJar.getName(), new LocalizeResource(sparkAssemblyJar));
+    // Localize the spark-assembly and its dependency jars
+    Set<File> sparkAssemblyJar = SparkUtils.locateSparkDependencyJars();
+    Set<String> sparkDependenciesClassPath = new HashSet<>();
+    for (File file : sparkAssemblyJar) {
```

Comment:
Maybe better to move this block to `SparkUtils`. Something like `Map<String, LocalizeResources> getSparkLocalizeResources()`. Then the extra classpath is the key set of the resulting map. In the implementation, you might want to use a `LinkedHashMap` so that the keys are in the same order as you put so that you can make sure the spark assembly jar always comes first.